x-pack/packages/ml/nested_property/src/set_nested_property.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const setNestedProperty = (obj: Record<string, any>, accessor: string, value: any) => {
```

Comment:
looked into this a bit but it gets tricky quickly because of the unknown deep nesting. guess that's why we left it quite generic back when creating it. going to leave the function code as is in this PR but added some jest tests in [51d58f3](https://github.com/elastic/kibana/pull/147912/commits/51d58f3149aa2a10a3dcc07ea52f1abc1a202dad).